### PR TITLE
Integrate visualfsharp

### PR DESCRIPTION
**Pull request by rojepp**

### CODE REVIEW

**Comment by baronfel** [comment]
The travis build seems to be failing because we're running the test in fsharp\tests\core\quotes\test.fsx without the required data.  At the very top of the file we conditionally include cslib.dll which contains Class1, which is referenced on line 544 of test.fsx.  This dll is not build during the travis build, because during the merge from visualfsharp to this repo we deleted the build.bat file in fsharp\tests\core\quotes\ that used the C# compiler to build cslib.dll.

So in short, I think we either need to make a cross platform way to build that cslib.dll file, or we should change the condition on line 539 to skip this test.  I'd prefer if we could find some way to compile and use the file, because I believe that reflection should work on mono.

**Comment by rojepp** [comment]
Great analysis, thanks. So, given that the .bat file isn't there, I guess this test isn't run on AppVeyor at all? 

mcs could be used to compile the .cs file, but maybe that's a separate PR.

The conditional on line 539 used to say `#if Portable` instead of `#if FSHARP_CORE_31`.

**Comment by baronfel** [comment]
I'd agree with that assesment. The appveyor build is actually very lenient compared to the travis build.  It just calls build.bat in the root of the repo, and that file doesn't call any of the tests that travis does using (cd tests/fsharp/core; ./run-opt.sh).

So, long-term tasks to fix that up :D

Also, long-term task to fix this test by using mcs to compile the necessary dll for this breaking test.

Also, really long term task to evaluate the use of #if Portable and see if there are sections that are of actual use to run on mono?

**Comment by rojepp** [comment]
AppVeyor has a 40 minutes maximum per build, so I guess that's the reason.

**Comment by dsyme** [comment]
It would be good to run the same x-plat tests on AppVeyor

The historical reason is this isn't done is that AppVeyor is generally for windows, and this repo is not used for packaging a Windows release of F# (which is done from Microsoft\visualfsharp).

**Comment by dsyme** [comment]
This looks ready to go to me?

**Comment by dsyme** [comment]
Thanks for the great contribution!
